### PR TITLE
Implementation

### PR DESCRIPTION
**Pull request by arb**
This is the initial implementation of this object. I copied all the tests over from Good, added some others and cleaned up the general implementation. This will serve as version 1 of this object.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,45 @@
...
+  - `httpsAgents` - the list of httpsAgents to report socket information about. Can be a single https.Agent or an array of agents. Defaults to Https.globalAgent.
+
+The oppsy object is an EventEmitter so it exposes the same API(`.on` and `.emit`) as the Node EventEmitter object. After it is started, it emits an "ops" event after a set interval with the collected ops information as the event payload.
+
+#### oppsy.start(interval)
+
+Starts an Oppsy object collecting network and server information.
+- `interval` - number of seconds to wait between each data sampling.
+
+#### opps.stop()
```
Comment (cjihrig) [inline README.md]: `oppsy`?

File: lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+exports.makeContinuation = (predicate) => {
+
+    return (callback) => {
+
+        process.nextTick(() => {
```
Comment (cjihrig) [inline lib/utils.js]: Since you're targeting new versions of Node, you can do `process.nextTick(callback, null, predicate());`

**Comment by cjihrig** [comment]
LGTM with a couple comments. You may or may not want to make this a plugin. Your call.

File: lib/network.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+// Load modules
+
+const Http = require('http');
+const Https = require('https');
+
+const Hoek = require('hoek');
+const Items = require('items');
+
+class Network {
+    constructor(server/*, httpAgents, httpsAgents */) {
```
Comment (Marsup) [inline lib/network.js]: Why not declare those arguments ? It's not like it's dynamic afterwards.

File: lib/network.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+        this._requests = {};
+        this._responseTimes = {};
+        this._server = server;
+        this._httpAgents = [].concat(arguments[1] || Http.globalAgent);
+        this._httpAgents = [].concat(arguments[2] || Https.globalAgent);
+
+        this._server.on('request-internal', this._onRequest.bind(this));
```
Comment (Marsup) [inline lib/network.js]: Minor but I think you should bind once.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+language: node_js
+
+node_js:
+  - 4.0
+  - 4
+  - 5
+
+sudo: false
```
Comment (arb) [inline .travis.yml]: I just copied the travis file used in hapi core.

File: lib/index.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+          psdelay: ProcessMonitor.delay,
+          requests: this._networkMonitor.requests,
+          concurrents: this._networkMonitor.concurrents,
+          responseTimes: this._networkMonitor.responseTimes,
+          sockets: this._networkMonitor.sockets
+      };
+  }
+  start(interval) {
+
+      Hoek.assert(interval <= 2147483647, 'interval must be less than 2147483648');
```
Comment (arb) [inline lib/index.js]: I :heart: big-time.

File: lib/network.js (hunk 4 above)
Comment (arb) [inline lib/network.js]: You mean use `once` instead of `on`? Yeah that's probably a good idea.

File: lib/network.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+            });
+        }
+    }
+    _onResponse(request) {
+
+        const msec = Date.now() - request.info.received;
+        const port = request.connection.info.port;
+        const statusCode = request.response && request.response.statusCode;
+
+        this._responseTimes[port] = this._responseTimes[port] || { count: 0, total: 0, max: 0 };
```
Comment (Marsup) [inline lib/network.js]: This looks like a lot of repeating of that thing

File: lib/network.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+            callback(err, result);
+        });
+    }
+    responseTimes(callback) {
+
+        const ports = Object.keys(this._responseTimes);
+        const overview = {};
+        for (let i = 0; i < ports.length; ++i) {
+            const port = ports[i];
+            const count = Hoek.reach(this, '_responseTimes.' + port + '.count', { default: 1 });
```
Comment (Marsup) [inline lib/network.js]: Template string ?

File: lib/process.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+const internals = {};
+
+
+module.exports = internals;
+
+
+internals.delay = (callback) => {
+
+    const bench = new Hoek.Bench();
+    setImmediate(() => {
```
Comment (Marsup) [inline lib/process.js]: Better in one line no ?

File: lib/network.js (hunk 4 above)
Comment (Marsup) [inline lib/network.js]: Oh nevermind, I'm tired, I thought you bound twice the same method... :sob:

File: .travis.yml (hunk 5 above)
Comment (Marsup) [inline .travis.yml]: Relevant blog post : http://blog.travis-ci.com/2015-03-31-docker-default-on-the-way/

**Pull request merged**